src/components/List/List.tsx

```diff
@@ -1,168 +1,80 @@
-        item: PropTypes.oneOfType([PropTypes.object, PropTypes.string, PropTypes.number]),
-        sortable: PropTypes.bool,
-        sortHandleAlign: PropTypes.oneOf(['left', 'right']),
-        active: PropTypes.bool,
-        selected: PropTypes.bool,
-
-        renderItem: PropTypes.func,
-        onMouseMove: PropTypes.func,
-        onClick: PropTypes.func,
-    };
-    static defaultProps = {
-        renderItem: (item) => String(item),
-    };
-    ref = React.createRef();
-    onMouseMove = () => {
-        if (this.props.onMouseMove) {
-            this.props.onMouseMove(this.props.itemIndex, this.props.item);
-        }
-    };
-    onClick = () => {
-        if (this.props.onClick) {
-            this.props.onClick(this.props.item, this.props.itemIndex);
-        }
-    };
-    getRef = () => this.ref;
-    renderSortIcon() {
-        const {sortable} = this.props;
-        return sortable ? (
-            <div className={b('item-sort-icon')}>
-                <DragHandleIcon />
-            </div>
-        ) : null;
-    }
-    renderContent() {
-        const {renderItem, item, active, itemIndex} = this.props;
-        return <div className={b('item-content')}>{renderItem(item, active, itemIndex)}</div>;
-    }
-    render() {
-        const {item, style, sortable, sortHandleAlign, itemClassName, selected, active} =
-            this.props;
-        return (
-            <div
-                className={b(
-                    'item',
-                    {
-                        sortable,
-                        active,
-                        selected,
-                        inactive: item.disabled,
-                        'sort-handle-align': sortHandleAlign,
-                    },
-                    itemClassName,
-                )}
-                style={style}
-                onClick={this.onClick}
-                onMouseMove={this.onMouseMove}
-                ref={this.ref}
-            >
-                {this.renderSortIcon()}
-                {this.renderContent()}
-            </div>
-        );
-    }
-}
-
 const SortableListItem = sortableElement(ListItem);
 const SortableListContainer = sortableContainer(ListContainer, {withRef: true});
-
 const SortableSimpleContainer = sortableContainer(SimpleContainer);
 
-export const listDefaultProps = {
+type ListState<T> = {
+    items: ListProps<T>['items'];
+    filter: string;
+    activeItem?: number;
+    pageSize?: number;
+    sorting?: boolean;
+};
+
+export const listDefaultProps: Partial<ListProps<ListItemData<unknown>>> = {
     items: [],
     itemClassName: '',
-
     filterable: true,
     sortable: false,
     virtualized: true,
-
     deactivateOnLeave: true,
 };
 
-export class List extends React.Component {
-    static moveListElement(list, oldIndex, newIndex) {
+export class List<T = unknown> extends React.Component<ListProps<T>, ListState<T>> {
+    static defaultProps: Partial<ListProps<ListItemData<unknown>>> = listDefaultProps;
```

Comment:
Seems there should be T instead of unknown

Reply:
We will have an error `Static members cannot reference class type parameters.` if change `unknown` to `T`